components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/responsemode/provider/impl/QueryResponseModeProvider.java

```diff
@@ -154,4 +160,21 @@ public AuthResponseType getAuthResponseType() {
         return AuthResponseType.REDIRECTION;
     }
 
+
+    private void appendQueryParam(List<String> queryParams, String key, String value) {
+
+        String encodedValue = encodeValue(value);
+        queryParams.add(key + "=" + encodedValue);
+    }
+
+    private String encodeValue(String value) {
+
+        try {
+            return URLEncoder.encode(value, StandardCharsets.UTF_8.toString());
+        } catch (UnsupportedEncodingException e) {
+            // This exception will not be thrown as UTF-8 is always supported.
+            log.error("Error occurred while encoding the value: " + value, e);
+            return null;

```

Comment:
Returning null might case NPE when concat with strings.
Let's fix it in a later PR.

Reply:
This exception will not be thrown since UTF-8 is always supported.